hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -702,8 +702,9 @@ public boolean truncate(String src, long newLength, String clientName)
     RemoteMethod method = new RemoteMethod("truncate",
         new Class<?>[] {String.class, long.class, String.class},
         new RemoteParam(), newLength, clientName);
+    // Truncate can return true/false, so don't expect a result
```

Comment:
How about `setReplication`, `rename`, `delete`, `mkdirs`?

Reply:
SetReplication API has the same issue. I think `setReplication` API should call  `rpcClient.invokeAll(locations, method)` like `delete` API.  I have fix it in next commit. Please review it again.